internal/observability/event/telemetry_filter.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package event
+
+import (
+	"errors"
+	"reflect"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// protoFilter is a signature for a struct field validation test
+type protoFilter func(field reflect.StructField) bool
+
+// telemetryFilter checks a struct field should be included in observation telemetry data
+func telemetryFilter(field reflect.StructField) bool {
+	if field.Tag.Get("eventstream") == "observation" {
+		// log.Printf("%s is allowed by Tags (%s:\"%s\")", name, tag, tagValue)
+		return true
+	}
+	return false
+}
+
+// filterValue will preserve or zero a value based on if it is classed as observable
+func filterValue(fv reflect.Value, isObservable bool) error {
+	if isObservable {
+		return nil // let data persist to telemetry
+	}
+
+	// check for nil value (prevent panics)
+	if fv == reflect.ValueOf(nil) {
+		return nil
+	}
+
+	if fv.Kind() == reflect.Ptr {
+		fv = fv.Elem()
+	}
+
+	// check to see if it's an exported struct field
+	if !fv.CanSet() {
+		return nil
+	}
+
+	fv.SetZero()
+
+	return nil
+}
+
```

Comment:
nitpick: Comment on here `recurseStructureWithProtoFilter` would be really useful

Reply:
I decided to remove the error return instead, because it had become unnecessary in filterValue()

Reply:
the error seems to still be here...

Reply:
I'd love a comment on this recurseStructureWithProtoFilter as well.

Reply:
I've added a comment. Apologies for the confusion too, I was referring to the error return in filterValue()